src/shader_recompiler/backend/spirv/emit_spirv_image.cpp

```diff
@@ -566,13 +566,31 @@ Id EmitImageRead(EmitContext& ctx, IR::Inst* inst, const IR::Value& index, Id co
         LOG_WARNING(Shader_SPIRV, "Typeless image read not supported by host");
         return ctx.ConstantNull(ctx.U32[4]);
     }
-    return Emit(&EmitContext::OpImageSparseRead, &EmitContext::OpImageRead, ctx, inst, ctx.U32[4],
-                Image(ctx, index, info), coords, std::nullopt, std::span<const Id>{});
+    const auto [image, is_integer] = Image(ctx, index, info);
+    const Id result_type{is_integer ? ctx.U32[4] : ctx.F32[4]};
+    Id color{Emit(&EmitContext::OpImageSparseRead, &EmitContext::OpImageRead, ctx, inst,
+                  result_type, image, coords, std::nullopt, std::span<const Id>{})};
+    if (!is_integer) {
+        const Id color_r{ctx.OpBitcast(ctx.U32[1], ctx.OpCompositeExtract(ctx.F32[1], color, 0))};
+        const Id color_g{ctx.OpBitcast(ctx.U32[1], ctx.OpCompositeExtract(ctx.F32[1], color, 1))};
+        const Id color_b{ctx.OpBitcast(ctx.U32[1], ctx.OpCompositeExtract(ctx.F32[1], color, 2))};
+        const Id color_a{ctx.OpBitcast(ctx.U32[1], ctx.OpCompositeExtract(ctx.F32[1], color, 3))};
+        color = ctx.OpCompositeConstruct(ctx.U32[4], color_r, color_g, color_b, color_a);
```

Comment:
You should be able to just:
```suggestion
        color = ctx.OpBitcast(ctx.U32[4], color);
```